packages/router/src/lib/routes.ts

```diff
@@ -144,15 +145,36 @@ function toRoutes(rawRoutes: RawRoute[], files: Files): Route[] {
         ? toRoutes(rawRoute.children, files)
         : undefined;
     let module: (() => Promise<RouteExport>) | undefined = undefined;
+    let meta: { endpoint: string; endpointKey: string } | undefined = undefined;
 
     if (rawRoute.filename) {
       const isMarkdownFile = rawRoute.filename.endsWith('.md');
       module = isMarkdownFile
         ? toMarkdownModule(files[rawRoute.filename] as () => Promise<string>)
         : (files[rawRoute.filename] as () => Promise<RouteExport>);
+
+      const endpointKey = rawRoute.filename.replace(
+        /\.page\.ts$/,
+        ENDPOINT_EXTENSION
+      );
+
+      // get endpoint path
+      const rawEndpoint = rawRoute.filename
+        .replace(/\.page\.ts$/, '')
+        .split(APP_DIR)[1];
+
+      // replace periods, remove (index) paths
+      const endpoint = (rawEndpoint || '')
+        .replace(/\./g, '/')
+        .replace(/\/\(.*?\)$/, '/index');
```

Comment:
will this work in case we have multiple pathless layouts? e.g. `(home).page.ts`, `(auth).page.ts`, etc.

Reply:
Good catch. The parenthesis should be replaced with something more URL-friendly